iam/src/providers/githubForkedRepoProvider.ts

```diff
@@ -0,0 +1,126 @@
+      verifiedUserPayload = await verifyGithub(accessToken);
+      verifiedUserRepoPayload = await verifyUserGithubRepo(verifiedUserPayload, accessToken);
+    } catch (e) {
+      return { valid: false };
+    } finally {
+      valid =
+        verifiedUserPayload &&
+        verifiedUserRepoPayload &&
+        verifiedUserPayload.id &&
+        verifiedUserRepoPayload.forks_count >= 1
+          ? true
+          : false;
+    }
+
+    return {
+      valid: valid,
+      record: {
+        id: `${verifiedUserPayload.id}gte1Fork`,
+      },
+    };
+  }
+}
+
+const requestAccessToken = async (code: string): Promise<string> => {
+  const clientId = process.env.GITHUB_CLIENT_ID;
+  const clientSecret = process.env.GITHUB_CLIENT_SECRET;
+
+  // Exchange the code for an access token
+  const tokenRequest = await axios.post(
+    `https://github.com/login/oauth/access_token?client_id=${clientId}&client_secret=${clientSecret}&code=${code}`,
+    {},
+    {
+      headers: { Accept: "application/json" },
+    }
+  );
+  if (tokenRequest.status != 200) {
+    throw `Post for request returned status code ${tokenRequest.status} instead of the expected 200`;
+  }
+  const tokenResponse = tokenRequest.data as GithubTokenResponse;
+  return tokenResponse.access_token;
+};
+
+const verifyGithub = async (ghAccessToken: string): Promise<GithubFindMyUserResponse> => {
+  let userRequest;
+  try {
+    // Now that we have an access token fetch the user details
+    userRequest = await axios.get("https://api.github.com/user", {
+      headers: { Authorization: `token ${ghAccessToken}` },
+    });
+    return userRequest.data as GithubFindMyUserResponse;
+  } catch (e) {
+    if (userRequest.status != 200) {
+      throw `User GET request returned status code ${userRequest.status} instead of the expected 200`;
+    }
+  }
+};
+
+const verifyUserGithubRepo = async (
+  userData: GithubFindMyUserResponse,
+  ghAccessToken: string
+): Promise<GithubUserRepoResponseData> => {
+  let repoRequest: GithubRepoRequestResponse;
+
+  try {
+    // fetch user repo data
+    repoRequest = await axios.get(`https://api.github.com/users/${userData.login}/repos`, {
+      headers: { Authorization: `token ${ghAccessToken}` },
+    });
+  } catch (e) {
+    if (repoRequest.status != 200) {
+      throw `Repo GET request returned status code ${repoRequest.status} instead of the expected 200`;
+    }
+  }
+
+  // Returns an object containing first instance of a user's repo if it has been forked,
+  // or the last checked repo with no forks
+  const userRepoForksCheck = repoRequest.data.find((repo: GithubUserRepoResponseData): GithubUserRepoResponseData => {
+    // Check to see if the authenticated GH user is the same as the repo owner,
+    // if the repo is not a fork of another repo, and if the repo fork count is gte 1
+    if (userData.id === repo.owner.id && !repo.fork && repo.forks_count >= 1) {
+      return repo;
+    } else {
+      return repo;
+    }
```

Comment:
We have the same return on the `if` and `else` branch. Returning `true` in he if branch and `false` by default (or on the else branch) would solve the issue.